sematic/types/types/dataclass.py

```diff
@@ -100,6 +100,8 @@ def _can_cast_to_dataclass(from_type: Any, to_type: Any) -> Tuple[bool, Optional
 
 @register_to_json_encodable(DataclassKey)
 def _dataclass_to_json_encodable(value: Any, type_: Any) -> Any:
+    if value is None:
+        raise ValueError(f"Expected {type_}, got None")
```

Comment:
We don't do this check in other serializers. Is this necessary for a particular reason?

Reply:
Without this check, if you accidentally pass in a `None` for the value, the error message you get isn't very clear. This clarifies it. I discovered this the hard way when I was iterating on the code and tried to deserialize a `None` at one point.